src/eva/data/ioda_obs_space.py

```diff
@@ -97,6 +97,9 @@ def execute(self, data_collections):
                 # Read header part of the file to get coordinates
                 ds_groups = xr.Dataset()
 
+                # Save sensor_channels for later
+                sensor_channels = ds_header['nchans']
```

Comment:
I think this code might need to be protected to ensure it's not used when `nchans` is not in the file.

Reply:
Yeah so then the first one would need:
```
has_sensor_channels = False
if 'nchans' in ds_header.keys():
    sensor_channels = ds_header['nchans']
    has_sensor_channels = True
```
and then the later line:

```
if has_sensor_channels:
    ds['nchans'] = sensor_channels
```

Or some other similar solution

Reply:
And that works too.